File: C++/Problem_012.cpp
#include "stdafx.h"
#include <iostream>
#include <algorithm>
#include <vector>
#include <math.h>

/*Author: Christian Seely*/

long getNumFactors(int t_n)
{
  long nFactors = 0;
  double bound = std::sqrt(t_n) + 1;
  for (int i = 1; i < bound; i++)
  {
    if (t_n%i == 0) nFactors += 1;
  }
  return (nFactors * 2);
}


long problemTwelve()
{
  int i = 2;
  int sum;
  while (1)
  {
    sum = i * (i + 1) / 2; // Obtain triangle number.
    if (getNumFactors(sum) > 500) return sum;
    i++;
  }

}

int main()
{
  std::cout << problemTwelve() << std::endl;
  return 0;
}

File: Python/Problem_012.py
# Author: Christian Seely

import math
# Function to find the number of factors for some
# number num.
def numFactors(num):
    nFactors = 0
    # We only have to check up to the square root of the num.
    bound = int(math.sqrt(num)+1)
    for div in range(1, bound):
        if not (num % div):
            nFactors += 1
    # Factors come in pairs multiple answer by two.
    return nFactors*2


def problemTwelve():
    i = 2
    while True:
        sum = int(i*(i+1)/2) # Obtain triangle number.
        if numFactors(sum) > 500:
            return sum
        i += 1

print(problemTwelve())

